[JUST A TRY] Drag nodes to insert

cpunion opened the pull request:
@fritz-c I try to add a `dndDropTypes` to make the tree accept other dnd types, and process the insertion (IT DOES NOT WORK FINE), is it a way to make it flexible for support https://github.com/fritz-c/react-sortable-tree/issues/60?

Note:
With this commit, it can insert node into the tree, but can't move or insert again. I know it is not correctly processing the data. Just a try.

NourSammour commented:
awesome, did you use this library to handle you use case?
i have very similar task

cpunion commented:
@NourSammour No, can't do that. I Add DropTarget to every node to handle addChild, and then drag to right place. Not a good idea, just can use in my case.

NourSammour commented:
Hi @cpunion  :D
i made PR based on your one, if you interested
https://github.com/fritz-c/react-sortable-tree/pull/76/

cpunion commented:
@fritz-c Rebased on master and fixed insertion, please review. I moved some code from *endDrag* to *drop* to make drag source more lightweight and easy to extend. Now no need an *addItem* function to process insertion (See #76), and easy to extract a helper method. How about this?

cpunion commented:
Added a helper function.

cpunion commented:
Oh no. I broke the behavior of drop outside.

cpunion commented:
Fix all drag drop behavior, but it's ugly. No idea how to cancel drop graceful.

AaronHarris commented:
Any update? Will this be merged?

cpunion commented:
No, it's not good. Just for discussion, I close it now. Wait for a better way.
